doc/classes/ScriptExtension.xml

```diff
@@ -22,6 +22,11 @@
 			<description>
 			</description>
 		</method>
+		<method name="_get_class_icon_path" qualifiers="virtual const">
+			<return type="String" />
+			<description>
+			</description>
```

Comment:
Needs description.

Reply:
I'm for enforcing writing descriptions for every added documentation field, regardless if the class is documented overall or not, and whether anyone can write it. Leaving empty fields like that is the reason why we have so much undocumented stuff.

> in the same consistent style

You mean the Godot documentation style used everywhere?

Reply:
```suggestion
			Returns the path to the script's icon texture, or empty string if the script has no custom icon.
			</description>
```

Reply:
That's not what this method does though, it's what it should do in your implementation :)

Reply:
I'm not sure the `ScriptExtension` class should even be exposed? You should never interact with it from GDScript or C#, and even GDExtension dosen't directly interact with it (from GDExtension you register some callbacks via function in `gdextension_interface.h` which get used by `ScriptExtension`).

If we edit `register_engine_classes.cpp` and remove the line:

```c++
ClassDB::register_engine_class<ScriptExtension>();
```

... I think (although, I didn't test) that it should still work fine for it's purpose, except as an "unexposed" class.

And then I think (although, again I didn't test) that `godot --doctool` should try to generate docs for it.

Reply:
Er, sorry, I got myself mixed up! I was thinking of `ScriptInstanceExtension` which GDExtension developers don't directly interact with. For `ScriptExtension`, the GDExtension developer _does_ directly interact with the class, so this should remain exposed.